src/status_im2/subs/multiaccount.cljs

```diff
@@ -207,30 +207,43 @@
    (pos? (count (get multiaccount :images)))))
 
 (defn- replace-multiaccount-image-uri
-  [multiaccount port]
-  (let [public-key (:public-key multiaccount)
-        theme      (theme/get-theme)
-        images     (:images multiaccount)
-        images     (reduce (fn [acc current]
-                             (let [key-uid    (:keyUid current)
-                                   image-name (:type current)
-                                   uri        (image-server/get-account-image-uri {:port       port
-                                                                                   :public-key public-key
-                                                                                   :image-name image-name
-                                                                                   :key-uid    key-uid
-                                                                                   :theme      theme
-                                                                                   :ring?      true})]
-                               (conj acc (assoc current :uri uri))))
-                           []
-                           images)]
+  [multiaccount port font-file avatar-opts]
+  (let [{:keys [key-uid ens-name images]} multiaccount
+        avatar-opts (assoc avatar-opts :override-ring? (if ens-name false nil))
+        theme (theme/get-theme)
+
+        images
+        (reduce
+         (fn [acc current]
+           (let [key-uid    (:keyUid current)
+                 image-name (:type current)
+                 uri        (image-server/get-account-image-uri-fn (merge {:port       port
+                                                                           :image-name image-name
+                                                                           :key-uid    key-uid
+                                                                           :theme      theme}
+                                                                          avatar-opts))]
+             (conj acc (assoc current :fn uri))))
+         []
+         images)
+
+        images
+        (if (seq images)
+          images
+          [{:fn (image-server/get-initials-avatar-uri-fn (merge
+                                                          {:port      port
+                                                           :key-uid   key-uid
+                                                           :theme     theme
+                                                           :font-file font-file}
+                                                          avatar-opts))}])]
     (assoc multiaccount :images images)))
```

Comment:
Same as before, I'd suggest not to shadow variables, and I think it's possible to align the `let` bindings in pairs, just add a line break before the calls to `merge`

Reply:
this one here is actually a vector..
changed the format, but still not good..

Reply:
I think that `images` binding that is using a `reduce` is more complex than required, because it's reducing a sequence to an empty vector for all the items, so I think `mapv` is enough:

```clojure
(reduce
 (fn [acc current]
   (let [key-uid    (:keyUid current)
         image-name (:type current)
         uri        (image-server/get-account-image-uri-fn (merge {:port       port
                                                                   :image-name image-name
                                                                   :key-uid    key-uid
                                                                   :theme      theme}
                                                                  avatar-opts))]
     (conj acc (assoc current :fn uri))))
 []
 images)
 
 ;; to:
 
(mapv (fn [{key-uid :keyUid image-name :type :as image}]
        (let [uri-fn (image-server/get-account-image-uri-fn
                      (merge {:port       port
                              :image-name image-name
                              :key-uid    key-uid
                              :theme      theme}
                             avatar-opts))]
          (assoc image :fn uri-fn)))
      images)
```

What do you think?

Also, if we format a little this function, it could look like this:

```clojure
(defn- replace-multiaccount-image-uri
  [multiaccount port font-file avatar-opts]
  (let [{:keys [key-uid
                ens-name
                images]} multiaccount
        theme            (theme/get-theme)
        avatar-opts      (assoc avatar-opts :override-ring? (when ens-name false))
        images-with-uri  (mapv (fn [{key-uid :keyUid image-name :type :as image}]
                                 (let [uri-fn (image-server/get-account-image-uri-fn
                                               (merge {:port       port
                                                       :image-name image-name
                                                       :key-uid    key-uid
                                                       :theme      theme}
                                                      avatar-opts))]
                                   (assoc image :fn uri-fn)))
                               images)
        new-images       (if (seq images-with-uri)
                           images-with-uri
                           [{:fn (image-server/get-initials-avatar-uri-fn
                                  (merge {:port      port
                                          :key-uid   key-uid
                                          :theme     theme
                                          :font-file font-file}
                                         avatar-opts))}])]
    (assoc multiaccount :images new-images))) ;; Not sure about `new-images` var name
```

I think shadowing vars is not a good idea (naming vars sharing the same scope with the same name), so I tried also renaming vars.

Reply:
Thanks. Updated based on your remove. This is definitely more clear.